packages/core/src/lib/jdk/JdkInstaller.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+'use strict';
+
+import * as path from 'path';
+import util = require('../util');
+
+const JDK_VERSION = '8u265-b01';
+const JDK_DIR = `jdk${JDK_VERSION}`;
+const DOWNLOAD_JDK_BIN_ROOT = `https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk${JDK_VERSION}/`;
+const DOWNLOAD_JDK_SRC_ROOT = 'https://github.com/AdoptOpenJDK/openjdk-jdk8u/archive/';
+const JDK_BIN_VERSION = JDK_VERSION.replace('-', '');
+const JDK_FILE_NAME_MAC = `OpenJDK8U-jdk_x64_mac_hotspot_${JDK_BIN_VERSION}.tar.gz`;
+const JDK_FILE_NAME_WIN32 = `OpenJDK8U-jdk_x86-32_windows_hotspot_${JDK_BIN_VERSION}.zip`;
+const JDK_FILE_NAME_LINUX64 = `OpenJDK8U-jdk_x64_linux_hotspot_${JDK_BIN_VERSION}.tar.gz`;
+const JDK_SRC_ZIP = `jdk${JDK_VERSION}.zip`;
+
+/**
+ * Checks whether JDK 8 is installed. If not installed,
+ * download the binary and source code and decompress at
+ * path given by user.
```

Comment:
Do you mind adding a comment saying why we need to download the JDK source code? Otherwise I could see some well intentioned person coming along and removing it since it's unused.

Reply:
Good point! I've added a line about it